src/main/java/me/namuhuchutong/ladder/domain/Row.java

```diff
@@ -1,60 +1,80 @@
 package me.namuhuchutong.ladder.domain;
 
-import me.namuhuchutong.ladder.domain.wrapper.*;
 import me.namuhuchutong.ladder.domain.factory.ScaffoldFactory;
+import me.namuhuchutong.ladder.domain.wrapper.Movements;
+import me.namuhuchutong.ladder.domain.wrapper.Point;
 
 import java.util.ArrayList;
 import java.util.List;
 
 import static java.util.Collections.*;
-import static me.namuhuchutong.ladder.domain.wrapper.LadderExpressionEnum.*;
+import static me.namuhuchutong.ladder.domain.wrapper.LadderExpression.*;
+import static me.namuhuchutong.ladder.domain.wrapper.Movements.*;
 
 public class Row {
 
-    private final List<LadderExpressionEnum> values;
+    private final List<Point> values;
 
     public static Row from(int participants, ScaffoldFactory factory) {
-        List<LadderExpressionEnum> collect = initializeLadderRow();
-        collect.addAll(addScaffold(participants, factory));
-        return new Row(unmodifiableList(collect));
+        List<Point> initializedRow = initializeLadderRow();
+        initializedRow.addAll(addScaffold(participants, factory));
+        return new Row(unmodifiableList(initializedRow));
     }
 
-    private static List<LadderExpressionEnum> addScaffold(int participants, ScaffoldFactory factory) {
-        List<LadderExpressionEnum> result = new ArrayList<>();
+    private static List<Point> addScaffold(int participants, ScaffoldFactory factory) {
+        List<Point> result = new ArrayList<>();
         for (int i = 1; i < participants; i++) {
-            result.add(factory.createScaffold());
-            result.add(VERTICAL_BAR);
+            result.add(new Point(factory));
+            result.add(new Point(VERTICAL_BAR));
         }
         return result;
     }
 
-    private static List<LadderExpressionEnum> initializeLadderRow() {
-        List<LadderExpressionEnum> list = new ArrayList<>();
-        list.add(EMPTY_SPACE);
-        list.add(VERTICAL_BAR);
+    private static List<Point> initializeLadderRow() {
+        List<Point> list = new ArrayList<>();
+        list.add(new Point(EMPTY_SPACE));
+        list.add(new Point(VERTICAL_BAR));
         return list;
     }
 
-    public Row(List<LadderExpressionEnum> values) {
+    public Row(List<Point> values) {
         validateContinuousScaffold(values);
         this.values = values;
     }
 
-    private void validateContinuousScaffold(List<LadderExpressionEnum> values) {
+    private void validateContinuousScaffold(List<Point> values) {
         ContinuousScaffoldValidator validator = new ContinuousScaffoldValidator();
         values.stream()
-              .filter(expression -> !expression.equals(VERTICAL_BAR))
+              .filter(point -> !point.isVerticalBar())
```

Comment:
```suggestion
              .filter(point -> point.isNotVerticalBar())
```
`!`이런게 붙으면 한번 더 생각해야 하기때문에 개인적으론 가독성 저해시키고 오용하기 쉬운 패턴이라고 생각해요 🤔